package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1908,17 +1959,43 @@ class Angles(_Connection):
     _n_atoms = 3
 
 
-class Dihedrals(_Connection):
+class Dihedrals(_SymmetricConnection):
     """A connection between four sequential atoms"""
     attrname = 'dihedrals'
     singular = 'dihedrals'
     transplants = defaultdict(list)
     _n_atoms = 4
 
+def _check_connection_types(func):
+    """
+    Checks values passed to _Connection methods for appropriate number of 
+    atom indices and coerces them to tuples of ints.
+
+
+    .. versionadded:: 0.21.0
+
+
+    """
+    @functools.wraps(func)
+    def wrapper(self, values, *args, **kwargs):
+        if not all(len(x) == self._n_atoms 
+                and all(isinstance(y, (int, np.integer)) for y in x)
+                for x in values):
+            raise ValueError(("{} must be an iterable of tuples with {}"
+                            " atom indices").format(self.attrname,
+                            self._n_atoms))
+        clean = [tuple(v) for v in values]
+        return func(self, clean, *args, **kwargs)
+    return wrapper
 
 class Impropers(_Connection):
```

Comment:
So impropers are also symmetrical see: https://github.com/MDAnalysis/mdanalysis/issues/2386#issuecomment-556033474

Reply:
Righto, switched them back -- we now can't assume the third atom is the central one that others are bonded to, as test_top.TOPBase used to.